crates/sui-core/src/unit_tests/move_package_upgrade_tests.rs

```diff
@@ -1085,6 +1086,21 @@ async fn test_upgraded_types_in_one_txn() {
         .await;
 
     assert!(effects.status.is_ok(), "{:#?}", effects.status);
+
+    // verify that the types of events match
+    let e1_type = StructTag::from_str(&format!("{package_v2}::base::BModEvent")).unwrap();
+    let e2_type = StructTag::from_str(&format!("{package_v3}::base::CModEvent")).unwrap();
+
+    let event_digest = effects.events_digest.unwrap();
+    let mut events = runner
+        .authority_state
+        .get_transaction_events(&event_digest)
+        .unwrap()
+        .data;
+    events.sort_by(|a, b| a.type_.name.as_str().cmp(b.type_.name.as_str()));
```

Comment:
nit: `sort_by_key`
```suggestion
    events.sort_by_key(|a| a.type_.name);
```

Reply:
I tried that initially but I am hitting some borrow checker issues when using `sort_by_key` ...

Reply:
Try borrowing the type name, the key doesn't need to be taken by value.